src/components/search-results/search-results.test.tsx

```diff
@@ -0,0 +1,108 @@
+import { act, fireEvent, render, screen } from '@testing-library/react';
+import { BrowserRouter, MemoryRouter } from 'react-router-dom';
+import '@testing-library/jest-dom';
+import { expect, test, vi } from 'vitest';
+import { appContext } from '../../App-context';
+import { AppContextDefaultValue, resultsItemType } from '../../types/types';
+import SearchResults from './search-results';
+
+test('verify that the component renders the specified number of cards', async () => {
+  const mockedContext = {
+    isLoading: false,
+    resultsItemInfo: [
+      { title: 'test-title', description: 'test-description' },
+      { title: 'test-title', description: 'test-description' },
+    ],
+  } as AppContextDefaultValue;
+  render(
+    <appContext.Provider value={mockedContext}>
+      <MemoryRouter>
+        <SearchResults />
+      </MemoryRouter>
+    </appContext.Provider>
+  );
+  const cards = await screen.findAllByTestId('card');
+  expect(cards.length).toBe(2);
+});
+
+test('check that an appropriate message is displayed if no cards are present', () => {
+  const mockedContext = {
+    isLoading: false,
+    resultsItemInfo: [] as resultsItemType,
+  } as AppContextDefaultValue;
+  render(
+    <appContext.Provider value={mockedContext}>
+      <MemoryRouter>
+        <SearchResults />
+      </MemoryRouter>
+    </appContext.Provider>
+  );
+
+  expect(
+    screen.getByText(/There are no results for this request/i)
```

Comment:
In this case, **expect** is optional. In the case of **_getBy_** if the element is not found the test will fail with an error